src/sqlfluff/utils/reflow/classes.py

```diff
@@ -0,0 +1,283 @@
+                    # code.
+                    if next_raw_slice and next_raw_slice[0].slice_type != "literal":
+                        reflow_logger.debug(
+                            "    !! TW: Skipping apparent trailing whitespace "
+                            "followed by template"
+                        )
+                        continue
+
+                fixes.append(LintFix("delete", trailing_whitespace))
+                reflow_logger.debug("    !! TW: TRAILING WHITESPACE")
+        return fixes
+
+
+@dataclass
+class ReflowSequence:
+    """Class for keeping track of elements in a reflow operation.
+
+    It is assumed that there will be alternating blocks and points
+    (even if some points have no segments). This is validated on
+    construction.
+
+    We assume points on each end because this is the case with a file.
+    """
+
+    elements: Sequence[Union[ReflowBlock, ReflowPoint]]
+    root_segment: BaseSegment
+
+    def __post_init__(self):
+        """Validate integrity"""
+        assert self.elements
+        # Check root segment is shared
+        assert all(elem.root_segment is self.root_segment for elem in self.elements)
+        # Check first and last
+        try:
+            if isinstance(self.elements[0], ReflowPoint):
+                # Check odds are all points
+                assert all(isinstance(elem, ReflowPoint) for elem in self.elements[::2])
+                # Check evens are all blocks
+                assert all(
+                    isinstance(elem, ReflowBlock) for elem in self.elements[1::2]
+                )
+            else:
+                # Check odds are all points
+                assert all(isinstance(elem, ReflowBlock) for elem in self.elements[::2])
+                # Check evens are all blocks
+                assert all(
+                    isinstance(elem, ReflowPoint) for elem in self.elements[1::2]
+                )
+        except AssertionError as err:
+            reflow_logger.exception(
+                "Assertion check on ReflowSequence failed. " "Elements: %s",
+                self.elements,
+            )
+            raise err
+
+    @classmethod
+    def from_raw_segments(
+        cls, segments: Sequence[RawSegment], root_segment: BaseSegment
+    ):
+        """Construct a ReflowSequence from a sequence of raw segments.
+
+        Aimed to be the basic constructor, which other more specific
+        ones may fall back to.
+        """
+        elem_buff = []
+        seg_buff = []
+        SeqClass: Union[None, Type[ReflowBlock], Type[ReflowPoint]] = None
+        for seg in segments:
+            NextClass: Union[Type[ReflowBlock], Type[ReflowPoint]]
+            # NOTE: Can indents be in a reflow block? I assume not.
+            # Placeholders certainly not because they also get indented.
+            if seg.is_type("whitespace", "newline", "end_of_file"):
+                NextClass = ReflowPoint
+            else:
+                NextClass = ReflowBlock
```

Comment:
```suggestion
            NextClass: Union[Type[ReflowBlock], Type[ReflowPoint]] = (
               ReflowPoint if seg.is_type("whitespace", "newline", "end_of_file")
               else ReflowBlock
            )
            # NOTE: Can indents be in a reflow block? I assume not.
            # Placeholders certainly not because they also get indented.

```